client/homebrew/brewRenderer/brewRenderer.jsx

```diff
@@ -149,13 +171,12 @@ const BrewRenderer = createClass({
 	renderPage : function(pageText, index){
 		let cleanPageText = this.sanitizeScriptTags(pageText);
 		if(this.props.renderer == 'legacy')
-			return <div className='phb page' id={`p${index + 1}`} dangerouslySetInnerHTML={{ __html: MarkdownLegacy.render(cleanPageText) }} key={index} />;
+			return <div className='pages' ref='pages' lang={`${this.props.lang || 'en'}`}><div className='phb page' id={`p${index + 1}`} dangerouslySetInnerHTML={{ __html: MarkdownLegacy.render(cleanPageText) }} key={index} /></div>;
 		else {
-			cleanPageText += `\n\n&nbsp;\n\\column\n&nbsp;`; //Artificial column break at page end to emulate column-fill:auto (until `wide` is used, when column-fill:balance will reappear)
+			cleanPageText = cleanPageText.split(/^\\page$/gm).join(`\n\n&nbsp;\n\\column\n&nbsp;\n\\page\n`); //Artificial column break at page end to emulate column-fill:auto (until `wide` is used, when column-fill:balance will reappear)
+
 			return (
-				<div className='page' id={`p${index + 1}`} key={index} >
-					<div className='columnWrapper' dangerouslySetInnerHTML={{ __html: Markdown.render(cleanPageText) }} />
-				</div>
+				<div className='pages' ref='pages' lang={`${this.props.lang || 'en'}`} dangerouslySetInnerHTML={{ __html: Markdown.render(cleanPageText || ' ', this.state.usePPR ? this.state.viewablePageNumber : -1, 3) }} />
```

Comment:
This logic seems to be duplicating the steps that `renderPages()` already does, of deciding which pages to render based on PPR being active.

Reply:
Effectively, yes - as V3 now renders everything as a single page, the V3 renderer needs to be passed the PPR information so it can prune the non-rendered pages.
So Legacy performs the PPR in `brewRenderer.jsx` while V3 performs PPR in `markdown.js`.